ImgShareData/ImgurApiSource.cs

```diff
@@ -177,12 +177,181 @@ protected internal void EnsureClientHasAuthorizationHeaders()
+        /// <param name="title"> The title of the album </param>
+        /// <param name="description">The description of the album</param>
+        /// <param name="albumPrivacy">Sets the privacy level of the album. Values are : public | hidden | secret. Defaults to user's privacy settings for logged in users.</param>
+        /// <param name="albumLayout">Sets the layout to display the album. Values are : blog | grid | horizontal | vertical</param>
+        /// <returns></returns>
+        public async Task<ImgurBasic> AlbumUpdateAsync(string albumID, List<string> imageIDs = null, string coverImageId = "", String title = "", String description = "", Privacy albumPrivacy = Privacy.ignore, Layout albumLayout = Layout.ignore)
+        {
+            if (albumID == null)
+            {
+                throw new ArgumentNullException("albumID");
+            }
+
+            MultipartFormDataContent content = new MultipartFormDataContent(BoundaryGuid.ToString());
+            if (imageIDs.Count != 0 || imageIDs == null)
+            {
+                string serializedImageList = await Task.Run(() => JsonConvert.SerializeObject(imageIDs));
+                content.Add(new StringContent(serializedImageList), ImgurEndpoints.ImageEndpointParameterLookup[ImgurParameters.ids]);
+            }
+            if (coverImageId != String.Empty && coverImageId != null && coverImageId != "")
+            {
+                content.Add(new StringContent(coverImageId), ImgurEndpoints.ImageEndpointParameterLookup[ImgurParameters.cover]);
+            }
+            if (title != "")
+            {
+                content.Add(new StringContent(title), ImgurEndpoints.ImageEndpointParameterLookup[ImgurParameters.title]);
+            }
+            if (description != "")
+            {
+                content.Add(new StringContent(title), ImgurEndpoints.ImageEndpointParameterLookup[ImgurParameters.description]);
+            }
+            if (albumPrivacy != Privacy.ignore)
+            {
+                content.Add(new StringContent(Utilities.convertToString(albumPrivacy)), ImgurEndpoints.ImageEndpointParameterLookup[ImgurParameters.privacy]);
+            }
+            if (albumLayout != Layout.ignore)
+            {
+                content.Add(new StringContent(Utilities.convertToString(albumLayout)), ImgurEndpoints.ImageEndpointParameterLookup[ImgurParameters.layout]);
+            }
+            String responseString = await PostAnonymousImgurDataAsync(ImgurEndpoints.AlbumUpdate(albumID), content);
+            ImgurBasic status = await Task.Run(() => JsonConvert.DeserializeObject<ImgurBasic>(responseString, _defaultSerializerSettings));
+            return status;
+        }
+
+        /// <summary>
+        /// Deletes an album
+        /// </summary>
+        /// <param name="albumID">The delete hash or album ID (only if logged in) you want to delete.</param>

```

Comment:
nit: is the more common scenerio (from usage data or api design or whatever) that this is likely to be an albumId or a deletehash - if deletehash is more common let's rename the param